daemon/containerd/image_snapshot.go

```diff
@@ -24,6 +25,22 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, parentIma
 		matcher = platforms.Only(*platform)
 	}
 
+	img, err := i.resolveImage(ctx, parentImage)
+	if err != nil {
+		return err
+	}
+	platformImg := containerd.NewImageWithPlatform(i.client, img, matcher)
+	unpacked, err := platformImg.IsUnpacked(ctx, i.snapshotter)
+	if err != nil {
+		return err
+	}
+
+	if !unpacked {
+		if err := platformImg.Unpack(ctx, i.snapshotter); err != nil {
+			return err
+		}
+	}
+
 	desc, err = containerdimages.Config(ctx, cs, desc, matcher)
```

Comment:
Silly question; does this step need the unpacked image to do its job? (Mostly wondering if the unpack can be done later)

Reply:
`Config` doesn't need the image to be unpacked, I can move the code but the descriptor returned by `Config` is immediately used to get the `RootFS` so I thought it might be nicer to bundle related things closer

Reply:
Gotcha; my take was "try to do lightweight checks first, before we do heavy-lifting".

I must admit that I haven't checked in-depth on that, but perhaps you know from the top of your head.

Reply:
`Config` is in fact a heavy-lifter :) 

I did however removed the call to `resolveDescriptor` and are calling `resolveImage` directly, that way we don't have to get the image twice